parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -1811,6 +1845,44 @@ public void readAll(SeekableInputStream f, ChunkListBuilder builder) throws IOEx
       }
     }
 
+    public void readAllVectored(SeekableInputStream f, ChunkListBuilder builder)
+      throws IOException, ExecutionException, InterruptedException {
+      LOG.warn("Reading through the vectored API.[readAllVectored]");
+      List<Chunk> result = new ArrayList<Chunk>(chunks.size());
+
+      int fullAllocations = (int) (length / options.getMaxAllocationSize());
+      int lastAllocationSize = (int) (length % options.getMaxAllocationSize());
+
+      int numAllocations = fullAllocations + (lastAllocationSize > 0 ? 1 : 0);
+      List<FileRange> fileRanges = new ArrayList<>(numAllocations);
+
+      long currentOffset = offset;
+      for (int i = 0; i < fullAllocations; i += 1) {
+        //buffers.add(options.getAllocator().allocate(options.getMaxAllocationSize()));
+        fileRanges.add(FileRange.createFileRange(currentOffset, options.getMaxAllocationSize()));
+        currentOffset = currentOffset + options.getMaxAllocationSize();
+      }
+
+      if (lastAllocationSize > 0) {
+        //buffers.add(options.getAllocator().allocate(lastAllocationSize));
+        fileRanges.add(FileRange.createFileRange(currentOffset, lastAllocationSize));
+      }
+      LOG.warn("Doing vectored IO for ranges {}", fileRanges);
+      f.readVectored(fileRanges, ByteBuffer::allocate);
```

Comment:
Use the allocator (`options.getAllocator`)? Keep in mind the allocated buffer might be a direct byte buffer.